Goobi/src/de/sub/goobi/metadaten/Metadaten.java

```diff
@@ -3098,15 +3099,59 @@ boolean canCreate(MetadataGroupType type) {
      *             multi-select input
      */
     public List<RenderableMetadataGroup> getMyGroups() throws ConfigurationException {
-        List<MetadataGroup> records;
-        if (myDocStruct == null || (records = myDocStruct.getAllMetadataGroups()) == null) {
+        if (myDocStruct == null) {
             return Collections.emptyList();
         }
+
+        List<MetadataGroup> records = myDocStruct.getAllMetadataGroups();
+        if (records == null) {
+            records = Collections.emptyList();
+        }
+        List<MetadataGroupType> defaultGroupTypes = myDocStruct.getType().getAllDefaultDisplayMetadataGroups();
+        if (records.isEmpty() && defaultGroupTypes.isEmpty()) {
+            return Collections.emptyList();
+        }
+
         List<RenderableMetadataGroup> result = new ArrayList<RenderableMetadataGroup>(records.size());
         String language = (String) Helper.getManagedBeanValue("#{LoginForm.myBenutzer.metadatenSprache}");
         String projectName = myProzess.getProjekt().getTitel();
+        Map<String, List<MetadataGroup>> displayGroups = myProzess.getRegelsatz().isOrderMetadataByRuleset()
+                ? new LinkedHashMap<String, List<MetadataGroup>>() : new TreeMap<String, List<MetadataGroup>>();
+        for (MetadataGroupType groupType : myDocStruct.getType().getAllMetadataGroupTypes()) {
+            displayGroups.put(groupType.getName(), null);
+        }
+        for (MetadataGroupType groupType : defaultGroupTypes) {
+            displayGroups.put(groupType.getName(), new ArrayList());
+        }
         for (MetadataGroup record : records) {
-            result.add(new RenderableMetadataGroup(record, this, language, projectName));
+            String name = record.getType().getName();
+            if (displayGroups.get(name) == null) {
+                displayGroups.put(name, new ArrayList());
+            }
+            displayGroups.get(name).add(record);
+        }
+        for (Entry<String, List<MetadataGroup>> entry : displayGroups.entrySet()) {
+            records = entry.getValue();
+            if (records != null) {
```

Comment:
can be combined with following if statement

Reply:
Yes, but then I just have to check for `records != null` twice. I do not see any improvement in that, but I changed it.